modules/meeting/app/controllers/recurring_meetings_controller.rb

```diff
@@ -168,18 +171,45 @@ def delete_scheduled
     redirect_to polymorphic_path([@project, @recurring_meeting]), status: :see_other
   end
 
-  def download_ics
-    ::RecurringMeetings::ICalService
-      .new(user: current_user, series: @recurring_meeting)
-      .call
+  def download_ics # rubocop:disable Metrics/AbcSize
+    service = ::RecurringMeetings::ICalService.new(user: current_user, series: @recurring_meeting)
+    result =
+      if params[:occurrence_id].present?
+        scheduled_meeting = @recurring_meeting.scheduled_meetings.find_by(meeting_id: params[:occurrence_id])
+        service.generate_occurrence(scheduled_meeting)
+      else
+        service.generate_series
+      end
+
+    result
       .on_failure { |call| render_500(message: call.message) }
       .on_success do |call|
       send_data call.result, filename: filename_for_content_disposition("#{@recurring_meeting.title}.ics")
```

Comment:
just a question: when we generate an ical for a single occurence, should we also include that info in the filename?

Reply:
Yes good point, I'll add that